frontend/demo/src/actions/conversationsFilter/index.ts

```diff
@@ -0,0 +1,88 @@
+
+import {StateModel} from '../../reducers';
+import {loadingConversationsAction} from '../conversations';
+
+export const RESET_FILTERED_CONVERSATIONS = '@@conversations/RESET_FILTEREDS';
+export const MERGE_FILTERED_CONVERSATIONS = '@@conversations/MERGE_FILTERED';
+export const UPDATE_CONVERSATION_FILTER = '@@conversation/update_filter';
+
+export const resetFilteredConversationAction = createAction(RESET_FILTERED_CONVERSATIONS);
+export const mergeFilteredConversationsAction = createAction(
+  MERGE_FILTERED_CONVERSATIONS,
+  resolve => (conversations: Conversation[], filter: ConversationFilter, metadata: ResponseMetadataPayload) =>
+    resolve({conversations, filter, metadata})
+);
+export const updateFilteredConversationsAction = createAction(
+  UPDATE_CONVERSATION_FILTER,
+  resolve => (filter: ConversationFilter) => resolve({filter})
+);
+
+export const setSearch = (currentFilter: ConversationFilter, displayName: string) => {
+  return setFilter({
+    ...currentFilter,
+    displayName,
+  });
+};
+
+export const setFilter = (filter: ConversationFilter) => {
+  return (dispatch: Dispatch<any>, state: () => StateModel) => {
+    executeFilter(filter, dispatch, state);
+  };
+};
+
+export const resetFilter = () => {
+  return function(dispatch: Dispatch<any>, state: () => StateModel) {
+    dispatch(resetFilteredConversationAction());
+    const currentFilter = state().data.conversations.filtered.currentFilter;
+    const newFilter: ConversationFilter = {displayName: currentFilter.displayName};
+    executeFilter(newFilter, dispatch, state);
+  };
+};
+
+const executeFilter = (filter: ConversationFilter, dispatch: Dispatch<any>, state: () => StateModel) => {
+  dispatch(updateFilteredConversationsAction(filter));
+  refetchConversations(dispatch, state);
+};
+
+const refetchConversations = (dispatch: Dispatch<any>, state: () => StateModel, cursor?: string) => {
+  dispatch(loadingConversationsAction());
+  const filter = state().data.conversations.filtered.currentFilter;
+  return HttpClientInstance.listConversations({
+    page_size: 10,
+    cursor,
+    filters: filterToLuceneSyntax(filter),
+  })
+    .then((response: {data: Conversation[]; metadata: ResponseMetadataPayload}) => {
+      dispatch(mergeFilteredConversationsAction(response.data, filter, response.metadata));
+      return Promise.resolve(true);
+    })
+    .catch((error: Error) => {
+      return Promise.reject(error);
+    });
+};
+
+const filterToLuceneSyntax = (filter: ConversationFilter): string | null => {
+  const filterQuery: Array<string> = [];
+  if (filter.unreadOnly) {
+    filterQuery.push('unread_count:[1 TO *]');
+  } else if (filter.readOnly) {
+    filterQuery.push('unread_count:0');
+  }
+  if (filter.displayName) {
+    filterQuery.push('display_name:*' + filter.displayName + '*');
+  }
+  if (filter.byTags && filter.byTags.length > 0) {
+    filterQuery.push('tag_id:(' + filter.byTags.join(' ') + ')');
```

Comment:
We will have to check if this works because we noted that UUIDs need to be escaped for the analyzer. On the backend tests we escape them like so: `.replace("-", "\\-")`

Reply:
this doesn't work but for a different reason: we have to `join(' AND ')`

Reply:
this was fixed :)